In [312]:
import pandas as pd
import numpy as np

### Reading Companies file

In [313]:
companies = pd.read_csv('companies.txt',sep='\t',encoding='iso-8859-1')
print(companies.shape)
companies

(66368, 10)


permalink  \
0                                    /Organization/-Fame   
1                                 /Organization/-Qounter   
2                    /Organization/-The-One-Of-Them-Inc-   
3                                  /Organization/0-6-Com   
4                         /Organization/004-Technologies   
5                       /Organization/01Games-Technology   
6                    /Organization/0Ndine-Biomedical-Inc   
7                                   /Organization/0Xdata   
8                                        /Organization/1   
9                              /Organization/1-2-3-Listo   
10                                 /Organization/1-4-All   
11                        /Organization/1-618-Technology   
12                           /Organization/1-800-Dentist   
13                           /Organization/1-800-Doctors   
14              /Organization/1-800-Publicrelations-Inc-   
15                            /Organization/1-Mainstream   
16                                 /Organization/1-Of-99   
17                             /Organization/10-20-Media   
18                         /Organization/10-Minutes-With   
19                              /Organization/1000-Corks   
20                            /Organization/1000-Markets   
21                                 /Organization/1000Chi   
22                               /Organization/1000Lookz   
23                            /Organization/1000Memories   
24                         /Organization/1000Museums-Com   
25                              /Organization/1001-Menus   
26                      /Organization/1001Pharmacies-Com   
27                                 /Organization/1006-Tv   
28                                /Organization/100Du-Tv   
29                                /Organization/100E-Com   
...                                                  ...   
66338                                /Organization/Zykis   
66339                       /Organization/Zylie-The-Bear   
66340                       /Organization/Zylun-Staffing   
66341                       /Organization/Zyme-Solutions   
66342                             /Organization/Zymergen   
66343                              /Organization/Zymetis   
66344                            /Organization/Zymeworks   
66345                             /Organization/Zymochem   
66346                                /Organization/Zympi   
66347                            /Organization/Zymr-Inc-   
66348                                /Organization/Zyncd   
66349                               /Organization/Zyncro   
66350              /Organization/Zynerba-Pharmaceuticals   
66351            /Organization/Zynergy-Projects-Services   
66352                                /Organization/Zynga   
66353                             /Organization/Zyngenia   
66354                              /Organization/Zynstra   
66355                           /Organization/Zyomyx-Inc   
66356                                 /Organization/Zype   
66357                               /Organization/Zypsee   
66358                     /Organization/Zyraz-Technology   
66359                                /Organization/Zyrra   
66360                               /Organization/Zystor   
66361                           /Organization/Zytoprotec   
66362                                /Organization/Zzish   
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                               /Organization/ÁEron   
66366                             /Organization/ÔAsys-2   
66367  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   

                                                name  \
0                                              #fame   
1                                           :Qounter   
2                             (THE) ONE of THEM,Inc.   
3                                            0-6.com   
4                                   004 Techno

### Handling encoding issues by removing special characters on all columns with datatype string

In [314]:
companies = companies[list(companies.columns)].applymap(lambda x:x.encode('utf-8').decode('ascii','ignore') if isinstance(x,str) else x)
print(companies.shape)
companies

(66368, 10)


permalink  \
0                                    /Organization/-Fame   
1                                 /Organization/-Qounter   
2                    /Organization/-The-One-Of-Them-Inc-   
3                                  /Organization/0-6-Com   
4                         /Organization/004-Technologies   
5                       /Organization/01Games-Technology   
6                    /Organization/0Ndine-Biomedical-Inc   
7                                   /Organization/0Xdata   
8                                        /Organization/1   
9                              /Organization/1-2-3-Listo   
10                                 /Organization/1-4-All   
11                        /Organization/1-618-Technology   
12                           /Organization/1-800-Dentist   
13                           /Organization/1-800-Doctors   
14              /Organization/1-800-Publicrelations-Inc-   
15                            /Organization/1-Mainstream   
16                                 /Organization/1-Of-99   
17                             /Organization/10-20-Media   
18                         /Organization/10-Minutes-With   
19                              /Organization/1000-Corks   
20                            /Organization/1000-Markets   
21                                 /Organization/1000Chi   
22                               /Organization/1000Lookz   
23                            /Organization/1000Memories   
24                         /Organization/1000Museums-Com   
25                              /Organization/1001-Menus   
26                      /Organization/1001Pharmacies-Com   
27                                 /Organization/1006-Tv   
28                                /Organization/100Du-Tv   
29                                /Organization/100E-Com   
...                                                  ...   
66338                                /Organization/Zykis   
66339                       /Organization/Zylie-The-Bear   
66340                       /Organization/Zylun-Staffing   
66341                       /Organization/Zyme-Solutions   
66342                             /Organization/Zymergen   
66343                              /Organization/Zymetis   
66344                            /Organization/Zymeworks   
66345                             /Organization/Zymochem   
66346                                /Organization/Zympi   
66347                            /Organization/Zymr-Inc-   
66348                                /Organization/Zyncd   
66349                               /Organization/Zyncro   
66350              /Organization/Zynerba-Pharmaceuticals   
66351            /Organization/Zynergy-Projects-Services   
66352                                /Organization/Zynga   
66353                             /Organization/Zyngenia   
66354                              /Organization/Zynstra   
66355                           /Organization/Zyomyx-Inc   
66356                                 /Organization/Zype   
66357                               /Organization/Zypsee   
66358                     /Organization/Zyraz-Technology   
66359                                /Organization/Zyrra   
66360                               /Organization/Zystor   
66361                           /Organization/Zytoprotec   
66362                                /Organization/Zzish   
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                                 /Organization/Eron   
66366                               /Organization/Asys-2   
66367  /Organization/Novatiff-Reklam-Ve-TanTM-Hizmetl...   

                                          name  \
0                                        #fame   
1                                     :Qounter   
2                       (THE) ONE of THEM,Inc.   
3                                      0-6.com   
4                             004 Technologies   
5                         

### Reading rounds2 file

In [315]:
rounds2 = pd.read_csv('rounds2.csv',encoding='iso-8859-1')
print(rounds2.shape)
rounds2

(114949, 6)


company_permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
5                          /ORGANIZATION/004-TECHNOLOGIES   
6                        /organization/01games-technology   
7                     /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8                     /organization/0ndine-biomedical-inc   
9                                    /ORGANIZATION/0XDATA   
10                                   /organization/0xdata   
11                                   /ORGANIZATION/0XDATA   
12                                   /organization/0xdata   
13                                        /ORGANIZATION/1   
14                                        /organization/1   
15                                        /ORGANIZATION/1   
16                              /organization/1-2-3-listo   
17                                  /ORGANIZATION/1-4-ALL   
18                         /organization/1-618-technology   
19                            /ORGANIZATION/1-800-DENTIST   
20                            /organization/1-800-doctors   
21               /ORGANIZATION/1-800-PUBLICRELATIONS-INC-   
22                             /organization/1-mainstream   
23                                  /ORGANIZATION/1-OF-99   
24                              /organization/10-20-media   
25                              /ORGANIZATION/10-20-MEDIA   
26                              /organization/10-20-media   
27                              /ORGANIZATION/10-20-MEDIA   
28                          /organization/10-minutes-with   
29                          /ORGANIZATION/10-MINUTES-WITH   
...                                                   ...   
114919                              /ORGANIZATION/ZYNSTRA   
114920                           /organization/zyomyx-inc   
114921                           /ORGANIZATION/ZYOMYX-INC   
114922                           /organization/zyomyx-inc   
114923                           /ORGANIZATION/ZYOMYX-INC   
114924                                 /organization/zype   
114925                                 /ORGANIZATION/ZYPE   
114926                               /organization/zypsee   
114927                               /ORGANIZATION/ZYPSEE   
114928                     /organization/zyraz-technology   
114929                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114930                     /organization/zyraz-technology   
114931                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114932                                /organization/zyrra   
114933                                /ORGANIZATION/ZYRRA   
114934                                /organization/zyrra   
114935                                /ORGANIZATION/ZYRRA   
114936                               /organization/zystor   
114937                           /ORGANIZATION/ZYTOPROTEC   
114938                                /organization/zzish   
114939                                /ORGANIZATION/ZZISH   
114940                                /organization/zzish   
114941  /ORGANIZATION/ZZNODE-SCIENCE-AND-TECHNOLOGY-CO...   
114942                          /organization/zzzzapp-com   
114943                          /ORGANIZATION/ZZZZAPP-COM   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink   funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633              venture   
2       /funding-

### Handling encoding issues by removing special characters on all columns with datatype string

In [316]:
rounds2 = rounds2[list(rounds2.columns)].applymap(lambda x:x.encode('utf-8').decode('ascii','ignore') if isinstance(x,str) else x)
print(rounds2.shape)
rounds2

(114949, 6)


company_permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
5                          /ORGANIZATION/004-TECHNOLOGIES   
6                        /organization/01games-technology   
7                     /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8                     /organization/0ndine-biomedical-inc   
9                                    /ORGANIZATION/0XDATA   
10                                   /organization/0xdata   
11                                   /ORGANIZATION/0XDATA   
12                                   /organization/0xdata   
13                                        /ORGANIZATION/1   
14                                        /organization/1   
15                                        /ORGANIZATION/1   
16                              /organization/1-2-3-listo   
17                                  /ORGANIZATION/1-4-ALL   
18                         /organization/1-618-technology   
19                            /ORGANIZATION/1-800-DENTIST   
20                            /organization/1-800-doctors   
21               /ORGANIZATION/1-800-PUBLICRELATIONS-INC-   
22                             /organization/1-mainstream   
23                                  /ORGANIZATION/1-OF-99   
24                              /organization/10-20-media   
25                              /ORGANIZATION/10-20-MEDIA   
26                              /organization/10-20-media   
27                              /ORGANIZATION/10-20-MEDIA   
28                          /organization/10-minutes-with   
29                          /ORGANIZATION/10-MINUTES-WITH   
...                                                   ...   
114919                              /ORGANIZATION/ZYNSTRA   
114920                           /organization/zyomyx-inc   
114921                           /ORGANIZATION/ZYOMYX-INC   
114922                           /organization/zyomyx-inc   
114923                           /ORGANIZATION/ZYOMYX-INC   
114924                                 /organization/zype   
114925                                 /ORGANIZATION/ZYPE   
114926                               /organization/zypsee   
114927                               /ORGANIZATION/ZYPSEE   
114928                     /organization/zyraz-technology   
114929                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114930                     /organization/zyraz-technology   
114931                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114932                                /organization/zyrra   
114933                                /ORGANIZATION/ZYRRA   
114934                                /organization/zyrra   
114935                                /ORGANIZATION/ZYRRA   
114936                               /organization/zystor   
114937                           /ORGANIZATION/ZYTOPROTEC   
114938                                /organization/zzish   
114939                                /ORGANIZATION/ZZISH   
114940                                /organization/zzish   
114941  /ORGANIZATION/ZZNODE-SCIENCE-AND-TECHNOLOGY-CO...   
114942                          /organization/zzzzapp-com   
114943                          /ORGANIZATION/ZZZZAPP-COM   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                                 /organization/eron   
114947                               /ORGANIZATION/ASYS-2   
114948  /organization/novatiff-reklam-ve-tantm-hizmetl...   

                                funding_round_permalink   funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633              venture   
2       /funding-

# Examining the quality of data

In [317]:
companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66367 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       51147 non-null object
dtypes: object(10)
memory usage: 5.1+ MB


In [318]:
rounds2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 6 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
dtypes: float64(1), object(5)
memory usage: 5.3+ MB


# DATA Cleaning

## Drop unwanted columns from Data frames
 
###### We will not use following columns for our analysis:
- funding_round_permalink    
- funding_round_code         
- funded_at                  
- name      
- homepage_url               
- status 
- state_code                 
- region     
- city 
- founded_at

In [319]:
companies.drop(['homepage_url',
             'status','state_code','region','city','founded_at'],axis=1,inplace=True)
companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 4 columns):
permalink        66368 non-null object
name             66367 non-null object
category_list    63220 non-null object
country_code     59410 non-null object
dtypes: object(4)
memory usage: 2.0+ MB


In [320]:
rounds2.drop(['funding_round_permalink','funding_round_code','funded_at'],axis=1,inplace=True)
rounds2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 3 columns):
company_permalink     114949 non-null object
funding_round_type    114949 non-null object
raised_amount_usd     94959 non-null float64
dtypes: float64(1), object(2)
memory usage: 2.6+ MB


### Lets check how many companies are avaible in companies column using 'permalink' column

In [321]:
len(list(companies['permalink'].str.lower().unique()))

66368

### There are 66368 companies. Lets check if we have same number of companies in round2 also?

In [322]:
len(list(rounds2['company_permalink'].str.lower().unique()))

66368

### Finding the unique key for merging dataframes

In [323]:
#Permalink column does not have null values. Lets check for duplicates. 
#While checking duplicates convert the column to either lower case or upper case so that we will not miss duplicates which are not in similar alphabetical case
companies[companies['permalink'].str.lower().duplicated() == True]

Empty DataFrame
Columns: [permalink, name, category_list, country_code]
Index: []

### In companies dataframe permalink column does not have null values and the values are unique with out duplicates so 'PERMALINK' column can be used as unique key.

In [324]:
# There are no null values in the company permalink column in the rounds2 database. Lets check for duplicates.
rounds2[rounds2['company_permalink'].str.lower().duplicated() == True]

company_permalink funding_round_type  \
2                    /organization/-qounter               seed   
8       /organization/0ndine-biomedical-inc            venture   
10                     /organization/0xdata            venture   
11                     /ORGANIZATION/0XDATA            venture   
12                     /organization/0xdata            venture   
14                          /organization/1        undisclosed   
15                          /ORGANIZATION/1               seed   
25                /ORGANIZATION/10-20-MEDIA     debt_financing   
26                /organization/10-20-media     debt_financing   
27                /ORGANIZATION/10-20-MEDIA     debt_financing   
29            /ORGANIZATION/10-MINUTES-WITH               seed   
35               /ORGANIZATION/1000MEMORIES               seed   
37            /ORGANIZATION/1000MUSEUMS-COM     debt_financing   
38            /organization/1000museums-com            venture   
39            /ORGANIZATION/1000MUSEUMS-COM            venture   
40            /organization/1000museums-com   convertible_note   
41            /ORGANIZATION/1000MUSEUMS-COM            venture   
42            /organization/1000museums-com            venture   
43            /ORGANIZATION/1000MUSEUMS-COM     debt_financing   
44            /organization/1000museums-com            venture   
46                 /organization/1001-menus            venture   
47                 /ORGANIZATION/1001-MENUS            venture   
48                 /organization/1001-menus               seed   
49                 /ORGANIZATION/1001-MENUS            venture   
52                    /organization/1006-tv            venture   
53                    /ORGANIZATION/1006-TV            venture   
55                   /ORGANIZATION/100DU-TV            venture   
57                   /ORGANIZATION/100E-COM            venture   
59                  /ORGANIZATION/100HEALTH            venture   
61                    /ORGANIZATION/100PLUS            venture   
...                                     ...                ...   
114899              /ORGANIZATION/ZYMR-INC-              angel   
114901                  /ORGANIZATION/ZYNCD               seed   
114903                 /ORGANIZATION/ZYNCRO            venture   
114907                  /ORGANIZATION/ZYNGA            venture   
114908                  /organization/zynga            venture   
114909                  /ORGANIZATION/ZYNGA            venture   
114910                  /organization/zynga            venture   
114911                  /ORGANIZATION/ZYNGA            venture   
114912                  /organization/zynga            venture   
114913                  /ORGANIZATION/ZYNGA              angel   
114914                  /organization/zynga            venture   
114917                /ORGANIZATION/ZYNSTRA            venture   
114918                /organization/zynstra               seed   
114919                /ORGANIZATION/ZYNSTRA              angel   
114921             /ORGANIZATION/ZYOMYX-INC              grant   
114922             /organization/zyomyx-inc            venture   
114923             /ORGANIZATION/ZYOMYX-INC            venture   
114925                   /ORGANIZATION/ZYPE               seed   
114927                 /ORGANIZATION/ZYPSEE               seed   
114929       /ORGANIZATION/ZYRAZ-TECHNOLOGY              angel   
114930       /organization/zyraz-technology              grant   
114931       /ORGANIZATION/ZYRAZ-TECHNOLOGY            venture   
114933                  /ORGANIZATION/ZYRRA               seed   
114934                  /organization/zyrra            venture   
114935                  /ORGANIZATION/ZYRRA              angel   
114939                  /ORGANIZATION/ZZISH              angel   
114940                  /organization/zzish               seed   
114943            /ORGANIZATION/ZZZZAPP-COM               seed   
114944            /organization/zzzzapp-com   convertible_note   


### We have same number of unique companies in round2 as in companies dataframe and the count is 66368. There are some duplicates in round2 table. There may be a chance that count may be equal but are companies in round2 that are not in companies dataframe.
### Lets check if the companies in round2 dataframe are also present in companies dataframe.

In [325]:
# To check the companies present in rounds2 which are not available in companies dataframe
rounds2.loc[~(rounds2['company_permalink'].str.lower().isin(companies['permalink'].str.lower()))]

Empty DataFrame
Columns: [company_permalink, funding_round_type, raised_amount_usd]
Index: []

### All the companies present in round2 dataframe are the same as the companies in companies dataframe. 

## Removing Null values
### We need to find the average amount invested in each funding type. So the rows with out amount(raised_amt_usd) can be ignored

In [326]:
# Ignoring rows with null values in raised_amount_usd column
rounds2 = rounds2[~rounds2['raised_amount_usd'].isnull()]
rounds2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 94959 entries, 0 to 114948
Data columns (total 3 columns):
company_permalink     94959 non-null object
funding_round_type    94959 non-null object
raised_amount_usd     94959 non-null float64
dtypes: float64(1), object(2)
memory usage: 2.9+ MB


### Formatting permalink and company_permalink in companies and round2 dataframes respectively to lower case so that we will have unique format across the columns that we use for merging.

In [327]:
rounds2_formatted = rounds2.copy()
companies_formatted = companies.copy()
rounds2_formatted['company_permalink'] = rounds2['company_permalink'].str.lower()
companies_formatted['permalink'] = companies['permalink'].str.lower()
companies_formatted.head()

permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
3                /organization/0-6-com                 0-6.com   
4       /organization/004-technologies        004 Technologies   

                                       category_list country_code  
0                                              Media          IND  
1  Application Platforms|Real Time|Social Network...          USA  
2                                  Apps|Games|Mobile          NaN  
3                                        Curated Web          CHN  
4                                           Software          USA

In [328]:
rounds2_formatted.head()

company_permalink funding_round_type  raised_amount_usd
0                  /organization/-fame            venture         10000000.0
2               /organization/-qounter               seed           700000.0
3  /organization/-the-one-of-them-inc-            venture          3406878.0
4                /organization/0-6-com            venture          2000000.0
6     /organization/01games-technology        undisclosed            41250.0

### Merged two dataframe to create master_frame for furthur Analysis

In [329]:
master_frame = pd.merge(rounds2_formatted,companies_formatted,left_on='company_permalink',right_on='permalink',how='left')
master_frame

company_permalink funding_round_type  \
0                                    /organization/-fame            venture   
1                                 /organization/-qounter               seed   
2                    /organization/-the-one-of-them-inc-            venture   
3                                  /organization/0-6-com            venture   
4                       /organization/01games-technology        undisclosed   
5                    /organization/0ndine-biomedical-inc               seed   
6                    /organization/0ndine-biomedical-inc            venture   
7                                   /organization/0xdata               seed   
8                                   /organization/0xdata            venture   
9                                   /organization/0xdata            venture   
10                                  /organization/0xdata            venture   
11                                       /organization/1               seed   
12                                       /organization/1               seed   
13                             /organization/1-2-3-listo               seed   
14                           /organization/1-800-doctors   convertible_note   
15              /organization/1-800-publicrelations-inc-     private_equity   
16                            /organization/1-mainstream            venture   
17                                 /organization/1-of-99   convertible_note   
18                             /organization/10-20-media     debt_financing   
19                             /organization/10-20-media     debt_financing   
20                             /organization/10-20-media     debt_financing   
21                             /organization/10-20-media     debt_financing   
22                         /organization/10-minutes-with            venture   
23                         /organization/10-minutes-with               seed   
24                              /organization/1000-corks               seed   
25                            /organization/1000-markets               seed   
26                                 /organization/1000chi              angel   
27                            /organization/1000memories            venture   
28                            /organization/1000memories               seed   
29                         /organization/1000museums-com     debt_financing   
...                                                  ...                ...   
94929                              /organization/zynstra            venture   
94930                              /organization/zynstra               seed   
94931                              /organization/zynstra              angel   
94932                           /organization/zyomyx-inc            venture   
94933                           /organization/zyomyx-inc              grant   
94934                           /organization/zyomyx-inc            venture   
94935                           /organization/zyomyx-inc            venture   
94936                                 /organization/zype               seed   
94937                                 /organization/zype               seed   
94938                               /organization/zypsee               seed   
94939                     /organization/zyraz-technology     private_equity   
94940                     /organization/zyraz-technology              angel   
94941                     /organization/zyraz-technology              grant   
94942                     /organization/zyraz-technology            venture   
94943                                /organization/zyrra               seed   
94944                                /organization/zyrra               seed   
94945                                /organization/zyrra            venture   
94946                                /organization/zyrra              angel   
94947                               /organization/zystor            venture   
94948     

In [330]:
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 94959 entries, 0 to 94958
Data columns (total 7 columns):
company_permalink     94959 non-null object
funding_round_type    94959 non-null object
raised_amount_usd     94959 non-null float64
permalink             94959 non-null object
name                  94958 non-null object
category_list         93915 non-null object
country_code          89108 non-null object
dtypes: float64(1), object(6)
memory usage: 5.8+ MB


### There are null values in the country code and category list, We can not remove the null values in these rows now as it may remove the rows that have valid funding type and raised amount which will impact the calculation of best funding type. In later steps we can eliminate these null values based on the requirement

In [331]:
#Extracting the funding types 

master_frame['funding_round_type'].unique()

array(['venture', 'seed', 'undisclosed', 'convertible_note',
       'private_equity', 'debt_financing', 'angel', 'grant',
       'equity_crowdfunding', 'post_ipo_equity', 'post_ipo_debt',
       'product_crowdfunding', 'secondary_market',
       'non_equity_assistance'], dtype=object)

### For the funding types, Calculate the average amount funded in millions

In [332]:
avg_fund = (master_frame.groupby('funding_round_type')['raised_amount_usd'].mean()/1000000).reset_index() #in millions
avg_fund

funding_round_type  raised_amount_usd
0                   angel           0.958694
1        convertible_note           1.453439
2          debt_financing          17.043526
3     equity_crowdfunding           0.538368
4                   grant           4.300576
5   non_equity_assistance           0.411203
6           post_ipo_debt         168.704572
7         post_ipo_equity          82.182494
8          private_equity          73.308593
9    product_crowdfunding           1.363131
10       secondary_market          79.649630
11                   seed           0.719818
12            undisclosed          19.242370
13                venture          11.748949

### Spark funds wants to invest between 5 to 15 million. Lets check which funding types are in this range.

In [333]:
avg_fund[(avg_fund['raised_amount_usd']  > 5) & (avg_fund['raised_amount_usd']  < 15)]

funding_round_type  raised_amount_usd
13            venture          11.748949

#### Funding round type : 'VENTURE' is the only type available in that range

In [334]:
tot_fund = (master_frame.groupby('funding_round_type')['raised_amount_usd'].sum()/1000000).reset_index() #in millions

print(tot_fund['raised_amount_usd'].sum())
tot_fund

990125.0851350001


funding_round_type  raised_amount_usd
0                   angel        4659.255123
1        convertible_note        2103.125572
2          debt_financing      113884.840886
3     equity_crowdfunding         661.116163
4                   grant        9134.424145
5   non_equity_assistance          30.017823
6           post_ipo_debt       25643.094917
7         post_ipo_equity       51610.606151
8          private_equity      141925.436105
9    product_crowdfunding         526.168593
10       secondary_market        2389.488903
11                   seed       16989.864181
12            undisclosed       30441.429697
13                venture      590126.216876

### For funding type - Venture, We need to calculate total funding for all countries and extract 9 countries that received highest funding.

# Removing Nulls
### From now we will be gathering data by grouping country, So the rows with out having country code can be ignored for analysis.

In [335]:
master_frame = master_frame[~master_frame['country_code'].isnull()].copy()
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 89108 entries, 0 to 94957
Data columns (total 7 columns):
company_permalink     89108 non-null object
funding_round_type    89108 non-null object
raised_amount_usd     89108 non-null float64
permalink             89108 non-null object
name                  89107 non-null object
category_list         88529 non-null object
country_code          89108 non-null object
dtypes: float64(1), object(6)
memory usage: 5.4+ MB


In [336]:
#Extracting top9 countries with total funding amount for funding_round_type venture is more
top9 = master_frame[master_frame['funding_round_type'] == 'venture'].groupby('country_code')['raised_amount_usd'].sum().reset_index().sort_values('raised_amount_usd',ascending=False).head(9)
top9

country_code  raised_amount_usd
94          USA       4.225108e+11
15          CHN       3.983542e+10
29          GBR       2.024563e+10
39          IND       1.439186e+10
12          CAN       9.583332e+09
28          FRA       7.259537e+09
42          ISR       6.907515e+09
21          DEU       6.346960e+09
45          JPN       3.363677e+09

USA - United States of America <br>
CHN - China <br>
GBR - United Kindom of Great Britain <br>
IND - India<br>
CAN - Canada<br>
FRA - France <br>
ISR - Israel <br>
DEU - Germany <br>
JPN - Japan<br>

### From the PDF given, The top three countries with english as official language are USA, GBR, IND

### Exracting the primary sector from category_list in master_frame and add it as column

In [337]:
master_frame['primary_sector'] = master_frame['category_list'].str.split('|').str[0]

In [338]:
master_frame

company_permalink funding_round_type  \
0                                    /organization/-fame            venture   
1                                 /organization/-qounter               seed   
3                                  /organization/0-6-com            venture   
4                       /organization/01games-technology        undisclosed   
5                    /organization/0ndine-biomedical-inc               seed   
6                    /organization/0ndine-biomedical-inc            venture   
7                                   /organization/0xdata               seed   
8                                   /organization/0xdata            venture   
9                                   /organization/0xdata            venture   
10                                  /organization/0xdata            venture   
11                                       /organization/1               seed   
12                                       /organization/1               seed   
13                             /organization/1-2-3-listo               seed   
14                           /organization/1-800-doctors   convertible_note   
15              /organization/1-800-publicrelations-inc-     private_equity   
16                            /organization/1-mainstream            venture   
17                                 /organization/1-of-99   convertible_note   
18                             /organization/10-20-media     debt_financing   
19                             /organization/10-20-media     debt_financing   
20                             /organization/10-20-media     debt_financing   
21                             /organization/10-20-media     debt_financing   
22                         /organization/10-minutes-with            venture   
23                         /organization/10-minutes-with               seed   
24                              /organization/1000-corks               seed   
25                            /organization/1000-markets               seed   
27                            /organization/1000memories            venture   
28                            /organization/1000memories               seed   
29                         /organization/1000museums-com     debt_financing   
30                         /organization/1000museums-com     debt_financing   
31                         /organization/1000museums-com            venture   
...                                                  ...                ...   
94928                              /organization/zynstra            venture   
94929                              /organization/zynstra            venture   
94930                              /organization/zynstra               seed   
94931                              /organization/zynstra              angel   
94932                           /organization/zyomyx-inc            venture   
94933                           /organization/zyomyx-inc              grant   
94934                           /organization/zyomyx-inc            venture   
94935                           /organization/zyomyx-inc            venture   
94936                                 /organization/zype               seed   
94937                                 /organization/zype               seed   
94938                               /organization/zypsee               seed   
94939                     /organization/zyraz-technology     private_equity   
94940                     /organization/zyraz-technology              angel   
94941                     /organization/zyraz-technology              grant   
94942                     /organization/zyraz-technology            venture   
94943                                /organization/zyrra               seed   
94944                                /organization/zyrra               seed   
94945                                /organization/zyrra            venture   
94946                                /organization/zyrra              angel   
94947     

### Extracting mapping.csv data to  dataframe

In [339]:
mapping = pd.read_csv('mapping.csv')
mapping

category_list  Automotive & Sports  Blanks  \
0                           NaN                    0       1   
1                            3D                    0       0   
2                   3D Printing                    0       0   
3                 3D Technology                    0       0   
4                    Accounting                    0       0   
5              Active Lifestyle                    0       0   
6                  Ad Targeting                    0       0   
7            Advanced Materials                    0       0   
8              Adventure Travel                    1       0   
9                   Advertising                    0       0   
10        Advertising Exchanges                    0       0   
11         Advertising Networks                    0       0   
12        Advertising Platforms                    0       0   
13                       Advice                    0       0   
14                    Aerospace                    1       0   
15                  Agriculture                    0       0   
16        Air Pollution Control                    0       0   
17                   Algorithms                    0       0   
18                  All Markets                    0       0   
19                 All Students                    0       0   
20          Alter0tive Medicine                    0       0   
21                       Alumni                    0       0   
22                     A0lytics                    0       0   
23                      Android                    0       0   
24                       Angels                    0       0   
25                  Animal Feed                    0       0   
26   Anything Capital Intensive                    0       0   
27                App Discovery                    0       0   
28                App Marketing                    0       0   
29                   App Stores                    0       0   
..                          ...                  ...     ...   
658                   Usability                    0       0   
659      User Experience Design                    0       0   
660              User Interface                    0       0   
661                   Utilities                    0       0   
662     Vending and Concessions                    0       0   
663             Venture Capital                    0       0   
664                   Veteri0ry                    0       0   
665                       Video                    0       0   
666          Video Conferencing                    0       0   
667                 Video Games                    0       0   
668             Video on Demand                    0       0   
669             Video Streaming                    0       0   
670          Virtual Workforces                    0       0   
671                        VoIP                    0       0   
672             Waste Ma0gement                    0       0   
673                       Watch                    0       0   
674                       Water                    0       0   
675          Water Purification                    0       0   
676                   Wearables                    0       0   
677                Web Browsers                    0       0   
678                  Web Design                    0       0   
679             Web Development                    0       0   
680                 Web Hosting                    0       0   
681                   Web Tools                    0       0   
682                    Weddings                    0       0   
683                   Wholesale                    0       0   
684            Wine And Spirits                    0       0   
685                    Wireless                    0       0   
686                       Women                    0       0   
687                Young Adults                    0       0   

     Cleantech / Semiconductors  Entertainment  H

In [340]:
mapping.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688 entries, 0 to 687
Data columns (total 10 columns):
category_list                              687 non-null object
Automotive & Sports                        688 non-null int64
Blanks                                     688 non-null int64
Cleantech / Semiconductors                 688 non-null int64
Entertainment                              688 non-null int64
Health                                     688 non-null int64
Manufacturing                              688 non-null int64
News, Search and Messaging                 688 non-null int64
Others                                     688 non-null int64
Social, Finance, Analytics, Advertising    688 non-null int64
dtypes: int64(9), object(1)
memory usage: 53.8+ KB


### There is null value in category list and this row is not required for our analysis. We have column called which is not a sector. So clean mapping to remove null values.

### In category_list column the data is missplet where 0 is there in place of na. Eg: A0lytics instead of Analytics. So need to correct.

In [341]:
mapping = mapping[~(mapping['category_list'].isnull())].drop(columns='Blanks') #Removeing nulls from category_list

# Replacing 0 with 'na'

mapping.loc[mapping['category_list'] != 'Enterprise 2.0', 'category_list'] = mapping.category_list.str.replace("0","na")
mapping

category_list  Automotive & Sports  \
1                                    3D                    0   
2                           3D Printing                    0   
3                         3D Technology                    0   
4                            Accounting                    0   
5                      Active Lifestyle                    0   
6                          Ad Targeting                    0   
7                    Advanced Materials                    0   
8                      Adventure Travel                    1   
9                           Advertising                    0   
10                Advertising Exchanges                    0   
11                 Advertising Networks                    0   
12                Advertising Platforms                    0   
13                               Advice                    0   
14                            Aerospace                    1   
15                          Agriculture                    0   
16                Air Pollution Control                    0   
17                           Algorithms                    0   
18                          All Markets                    0   
19                         All Students                    0   
20                 Alternative Medicine                    0   
21                               Alumni                    0   
22                            Analytics                    0   
23                              Android                    0   
24                               Angels                    0   
25                          Animal Feed                    0   
26           Anything Capital Intensive                    0   
27                        App Discovery                    0   
28                        App Marketing                    0   
29                           App Stores                    0   
30   Application Performance Monitoring                    0   
..                                  ...                  ...   
658                           Usability                    0   
659              User Experience Design                    0   
660                      User Interface                    0   
661                           Utilities                    0   
662             Vending and Concessions                    0   
663                     Venture Capital                    0   
664                          Veterinary                    0   
665                               Video                    0   
666                  Video Conferencing                    0   
667                         Video Games                    0   
668                     Video on Demand                    0   
669                     Video Streaming                    0   
670                  Virtual Workforces                    0   
671                                VoIP                    0   
672                    Waste Management                    0   
673                               Watch                    0   
674                               Water                    0   
675                  Water Purification                    0   
676                           Wearables                    0   
677                        Web Browsers                    0   
678                          Web Design                    0   
679                     Web Development                    0   
680                         Web Hosting                    0   
681                           Web Tools                    0   
682                            Weddings                    0   
683                           Wholesale                    0   
684                    Wine And Spirits                    0   
685                            Wireless                    0   
686                               Women                    0   
687                        Young Adults                    0   

     Cleantech / Semiconductors  Entertainment  Health  M

### create column in mapping dataframe to tag main_sector of each category in the category list which can be used for merging prmiary_sector in master frame and main_sector in mapping dataframe

In [342]:
mapping['main_sector'] = (mapping == 1).idxmax(axis=1)
mapping

category_list  Automotive & Sports  \
1                                    3D                    0   
2                           3D Printing                    0   
3                         3D Technology                    0   
4                            Accounting                    0   
5                      Active Lifestyle                    0   
6                          Ad Targeting                    0   
7                    Advanced Materials                    0   
8                      Adventure Travel                    1   
9                           Advertising                    0   
10                Advertising Exchanges                    0   
11                 Advertising Networks                    0   
12                Advertising Platforms                    0   
13                               Advice                    0   
14                            Aerospace                    1   
15                          Agriculture                    0   
16                Air Pollution Control                    0   
17                           Algorithms                    0   
18                          All Markets                    0   
19                         All Students                    0   
20                 Alternative Medicine                    0   
21                               Alumni                    0   
22                            Analytics                    0   
23                              Android                    0   
24                               Angels                    0   
25                          Animal Feed                    0   
26           Anything Capital Intensive                    0   
27                        App Discovery                    0   
28                        App Marketing                    0   
29                           App Stores                    0   
30   Application Performance Monitoring                    0   
..                                  ...                  ...   
658                           Usability                    0   
659              User Experience Design                    0   
660                      User Interface                    0   
661                           Utilities                    0   
662             Vending and Concessions                    0   
663                     Venture Capital                    0   
664                          Veterinary                    0   
665                               Video                    0   
666                  Video Conferencing                    0   
667                         Video Games                    0   
668                     Video on Demand                    0   
669                     Video Streaming                    0   
670                  Virtual Workforces                    0   
671                                VoIP                    0   
672                    Waste Management                    0   
673                               Watch                    0   
674                               Water                    0   
675                  Water Purification                    0   
676                           Wearables                    0   
677                        Web Browsers                    0   
678                          Web Design                    0   
679                     Web Development                    0   
680                         Web Hosting                    0   
681                           Web Tools                    0   
682                            Weddings                    0   
683                           Wholesale                    0   
684                    Wine And Spirits                    0   
685                            Wireless                    0   
686                               Women                    0   
687                        Young Adults                    0   

     Cleantech / Semiconductors  Entertainment  Health  M

### Merging master data frame with mapping data frame to add main_sector to the master_frame.

In [343]:
master_frame = pd.merge(master_frame,mapping[['category_list','main_sector']],left_on='primary_sector',right_on='category_list',how='left')
master_frame = master_frame.drop(columns=['category_list_y'])
master_frame = master_frame.rename(columns = {'category_list_x':'category_list'})
master_frame

company_permalink funding_round_type  \
0                                    /organization/-fame            venture   
1                                 /organization/-qounter               seed   
2                                  /organization/0-6-com            venture   
3                       /organization/01games-technology        undisclosed   
4                    /organization/0ndine-biomedical-inc               seed   
5                    /organization/0ndine-biomedical-inc            venture   
6                                   /organization/0xdata               seed   
7                                   /organization/0xdata            venture   
8                                   /organization/0xdata            venture   
9                                   /organization/0xdata            venture   
10                                       /organization/1               seed   
11                                       /organization/1               seed   
12                             /organization/1-2-3-listo               seed   
13                           /organization/1-800-doctors   convertible_note   
14              /organization/1-800-publicrelations-inc-     private_equity   
15                            /organization/1-mainstream            venture   
16                                 /organization/1-of-99   convertible_note   
17                             /organization/10-20-media     debt_financing   
18                             /organization/10-20-media     debt_financing   
19                             /organization/10-20-media     debt_financing   
20                             /organization/10-20-media     debt_financing   
21                         /organization/10-minutes-with            venture   
22                         /organization/10-minutes-with               seed   
23                              /organization/1000-corks               seed   
24                            /organization/1000-markets               seed   
25                            /organization/1000memories            venture   
26                            /organization/1000memories               seed   
27                         /organization/1000museums-com     debt_financing   
28                         /organization/1000museums-com     debt_financing   
29                         /organization/1000museums-com            venture   
...                                                  ...                ...   
89078                              /organization/zynstra            venture   
89079                              /organization/zynstra            venture   
89080                              /organization/zynstra               seed   
89081                              /organization/zynstra              angel   
89082                           /organization/zyomyx-inc            venture   
89083                           /organization/zyomyx-inc              grant   
89084                           /organization/zyomyx-inc            venture   
89085                           /organization/zyomyx-inc            venture   
89086                                 /organization/zype               seed   
89087                                 /organization/zype               seed   
89088                               /organization/zypsee               seed   
89089                     /organization/zyraz-technology     private_equity   
89090                     /organization/zyraz-technology              angel   
89091                     /organization/zyraz-technology              grant   
89092                     /organization/zyraz-technology            venture   
89093                                /organization/zyrra               seed   
89094                                /organization/zyrra               seed   
89095                                /organization/zyrra            venture   
89096                                /organization/zyrra              angel   
89097     

In [344]:
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 89108 entries, 0 to 89107
Data columns (total 9 columns):
company_permalink     89108 non-null object
funding_round_type    89108 non-null object
raised_amount_usd     89108 non-null float64
permalink             89108 non-null object
name                  89107 non-null object
category_list         88529 non-null object
country_code          89108 non-null object
primary_sector        88529 non-null object
main_sector           88219 non-null object
dtypes: float64(1), object(8)
memory usage: 6.8+ MB


### There are null values in the main_sector, As we are doing sector wise analysis we can ignore the rows with null values in this field. So removing rows with null values in main_sector

In [345]:
#ignoring rows with null values in main_sector
master_frame = master_frame[~master_frame['main_sector'].isnull()].copy()
master_frame.info()
master_frame

<class 'pandas.core.frame.DataFrame'>
Int64Index: 88219 entries, 0 to 89107
Data columns (total 9 columns):
company_permalink     88219 non-null object
funding_round_type    88219 non-null object
raised_amount_usd     88219 non-null float64
permalink             88219 non-null object
name                  88218 non-null object
category_list         88219 non-null object
country_code          88219 non-null object
primary_sector        88219 non-null object
main_sector           88219 non-null object
dtypes: float64(1), object(8)
memory usage: 6.7+ MB


company_permalink funding_round_type  \
0                                    /organization/-fame            venture   
1                                 /organization/-qounter               seed   
2                                  /organization/0-6-com            venture   
3                       /organization/01games-technology        undisclosed   
4                    /organization/0ndine-biomedical-inc               seed   
5                    /organization/0ndine-biomedical-inc            venture   
6                                   /organization/0xdata               seed   
7                                   /organization/0xdata            venture   
8                                   /organization/0xdata            venture   
9                                   /organization/0xdata            venture   
10                                       /organization/1               seed   
11                                       /organization/1               seed   
12                             /organization/1-2-3-listo               seed   
13                           /organization/1-800-doctors   convertible_note   
14              /organization/1-800-publicrelations-inc-     private_equity   
15                            /organization/1-mainstream            venture   
16                                 /organization/1-of-99   convertible_note   
17                             /organization/10-20-media     debt_financing   
18                             /organization/10-20-media     debt_financing   
19                             /organization/10-20-media     debt_financing   
20                             /organization/10-20-media     debt_financing   
21                         /organization/10-minutes-with            venture   
22                         /organization/10-minutes-with               seed   
23                              /organization/1000-corks               seed   
24                            /organization/1000-markets               seed   
25                            /organization/1000memories            venture   
26                            /organization/1000memories               seed   
27                         /organization/1000museums-com     debt_financing   
28                         /organization/1000museums-com     debt_financing   
29                         /organization/1000museums-com            venture   
...                                                  ...                ...   
89078                              /organization/zynstra            venture   
89079                              /organization/zynstra            venture   
89080                              /organization/zynstra               seed   
89081                              /organization/zynstra              angel   
89082                           /organization/zyomyx-inc            venture   
89083                           /organization/zyomyx-inc              grant   
89084                           /organization/zyomyx-inc            venture   
89085                           /organization/zyomyx-inc            venture   
89086                                 /organization/zype               seed   
89087                                 /organization/zype               seed   
89088                               /organization/zypsee               seed   
89089                     /organization/zyraz-technology     private_equity   
89090                     /organization/zyraz-technology              angel   
89091                     /organization/zyraz-technology              grant   
89092                     /organization/zyraz-technology            venture   
89093                                /organization/zyrra               seed   
89094                                /organization/zyrra               seed   
89095                                /organization/zyrra            venture   
89096                                /organization/zyrra              angel   
89097     

### Merged Dataframe with primary sector and mainsector

In [346]:

merged_df = master_frame[['primary_sector','main_sector']]
merged_df = merged_df.drop_duplicates('primary_sector')
merged_df

primary_sector                              main_sector
0                         Media                            Entertainment
1         Application Platforms               News, Search and Messaging
2                   Curated Web               News, Search and Messaging
3                         Games                            Entertainment
4                 Biotechnology               Cleantech / Semiconductors
6                     Analytics  Social, Finance, Analytics, Advertising
10                       Mobile               News, Search and Messaging
12                   E-Commerce                                   Others
13          Health and Wellness                                   Health
14           Internet Marketing  Social, Finance, Analytics, Advertising
15                         Apps               News, Search and Messaging
16                Entertainment                            Entertainment
21                    Education                                   Others
23                       Search               News, Search and Messaging
24                          Art                            Entertainment
34             Local Businesses                                   Others
39                    Cosmetics                                   Health
41                  Hospitality                                   Others
44                  Health Care                                   Health
48                     Software                                   Others
51          Enterprise Software                                   Others
57                Social Travel  Social, Finance, Analytics, Advertising
60                        Audio                            Entertainment
63           Health Diagnostics                                   Health
72            Service Providers                                   Others
73                   Publishing               News, Search and Messaging
75               Consumer Goods                            Manufacturing
76                Manufacturing                            Manufacturing
85            Augmented Reality  Social, Finance, Analytics, Advertising
86                  Advertising  Social, Finance, Analytics, Advertising
...                         ...                                      ...
62885              Music Venues                            Entertainment
64733  Independent Music Labels                            Entertainment
64747          Search Marketing               News, Search and Messaging
65797               Moneymaking  Social, Finance, Analytics, Advertising
66238    Independent Pharmacies                                   Health
67180          Cosmetic Surgery                                   Health
68619     Mobile Infrastructure                            Manufacturing
69699     Enterprise Purchasing  Social, Finance, Analytics, Advertising
70265                   mHealth                                   Health
71214           Shared Services                                   Others
71302               Promotional  Social, Finance, Analytics, Advertising
71389             Personal Data                                   Others
71431              Young Adults                                   Others
74834                   Surveys                                   Others
74931        Water Purification               Cleantech / Semiconductors
75660           Social Business  Social, Finance, Analytics, Advertising
76018    User Experience Design                                   Others
77128                   Indians                                   Others
77866             Social Buying  Social, Finance, Analytics, Advertising
77935   Social + Mobile + Local  Social, Finance, Analytics, Advertising
79752              Ride Sharing               News, Search and Messaging
80926                 Musicians                            Entertainment
81603                 Usability                                   Others
81

### Creating Datasets for three countries that we found in previous steps for funding round type - Venture and in investment range 5-15M

In [347]:
D1 = master_frame[(master_frame['country_code'] == 'USA') & (master_frame['funding_round_type'] == 'venture') & 
              (master_frame['raised_amount_usd'] > 5000000) & (master_frame['raised_amount_usd'] < 15000000)].copy()
D2 = master_frame[(master_frame['country_code'] == 'GBR') & (master_frame['funding_round_type'] == 'venture') & 
              (master_frame['raised_amount_usd'] > 5000000) & (master_frame['raised_amount_usd'] < 15000000)].copy()
D3 = master_frame[(master_frame['country_code'] == 'IND') & (master_frame['funding_round_type'] == 'venture') & 
              (master_frame['raised_amount_usd'] > 5000000) & (master_frame['raised_amount_usd'] < 15000000)].copy()

In [348]:
D1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9942 entries, 9 to 89097
Data columns (total 9 columns):
company_permalink     9942 non-null object
funding_round_type    9942 non-null object
raised_amount_usd     9942 non-null float64
permalink             9942 non-null object
name                  9942 non-null object
category_list         9942 non-null object
country_code          9942 non-null object
primary_sector        9942 non-null object
main_sector           9942 non-null object
dtypes: float64(1), object(8)
memory usage: 776.7+ KB


In [349]:
D2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 559 entries, 291 to 89078
Data columns (total 9 columns):
company_permalink     559 non-null object
funding_round_type    559 non-null object
raised_amount_usd     559 non-null float64
permalink             559 non-null object
name                  559 non-null object
category_list         559 non-null object
country_code          559 non-null object
primary_sector        559 non-null object
main_sector           559 non-null object
dtypes: float64(1), object(8)
memory usage: 43.7+ KB


In [350]:
D3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 243 entries, 0 to 88878
Data columns (total 9 columns):
company_permalink     243 non-null object
funding_round_type    243 non-null object
raised_amount_usd     243 non-null float64
permalink             243 non-null object
name                  243 non-null object
category_list         243 non-null object
country_code          243 non-null object
primary_sector        243 non-null object
main_sector           243 non-null object
dtypes: float64(1), object(8)
memory usage: 19.0+ KB


### From these data sets we need to do sector wise analysis 

In [351]:
#D1.groupby('main_sector')['raised_amount_usd'].sum() # Total Sum for different mainsectors
#D1.groupby('main_sector')['raised_amount_usd'].count() # Total count for different mainsectors

# By Transposing above dataframes we can create sum and count columns in D1

D1['Totval'] = D1.groupby('main_sector')['raised_amount_usd'].transform('sum')
D1['Totcount'] = D1.groupby('main_sector')['raised_amount_usd'].transform('count')
D1

company_permalink funding_round_type  \
9                       /organization/0xdata            venture   
72              /organization/128-technology            venture   
80           /organization/1366-technologies            venture   
81           /organization/1366-technologies            venture   
82           /organization/1366-technologies            venture   
95                 /organization/170-systems            venture   
102                    /organization/17zuoye            venture   
149                 /organization/2080-media            venture   
154                 /organization/2080-media            venture   
166                       /organization/21e6            venture   
173         /organization/22nd-century-group            venture   
182                    /organization/23andme            venture   
185                    /organization/23andme            venture   
187                    /organization/23andme            venture   
210                     /organization/28msec            venture   
239                  /organization/2nd-watch            venture   
250                         /organization/2u            venture   
252                         /organization/2u            venture   
262            /organization/3-v-biosciences            venture   
264            /organization/3-v-biosciences            venture   
267            /organization/3-v-biosciences            venture   
273                   /organization/33across            venture   
275                   /organization/33across            venture   
278                /organization/360commerce            venture   
312        /organization/3crowd-technologies            venture   
331                /organization/3d-robotics            venture   
362           /organization/3dr-laboratories            venture   
378                      /organization/3leaf            venture   
387             /organization/3pillar-global            venture   
389             /organization/3pillar-global            venture   
...                                      ...                ...   
88899      /organization/zounds-hearing-aids            venture   
88907                     /organization/zozi            venture   
88912                   /organization/zpower            venture   
88914                   /organization/zpower            venture   
88915                   /organization/zpower            venture   
88939                /organization/zuberance            venture   
88941                    /organization/zubie            venture   
88942                    /organization/zubie            venture   
88946                     /organization/zudy            venture   
88962                   /organization/zulily            venture   
88972                   /organization/zumbox            venture   
88974                   /organization/zumbox            venture   
88981                   /organization/zumigo            venture   
88984                   /organization/zumobi            venture   
88985                   /organization/zumper            venture   
88987                   /organization/zumper            venture   
88990          /organization/zuniversity-com            venture   
88992                    /organization/zuora            venture   
89004             /organization/zurex-pharma            venture   
89019                   /organization/zvents            venture   
89034          /organization/zyga-technology            venture   
89040          /organization/zyga-technology            venture   
89042         /organization/zygo-corporation            venture   
89068  /organization/zynerba-pharmaceuticals            venture   
89070                    /organization/zynga            venture   
89071                    /organization/zynga            venture   
89076                    /organization/zynga            venture   
89082               /organization/zyomyx-inc            venture   
89084  

In [352]:
# Adding the sum and count as columns to D2
D2['Totval'] = D2.groupby('main_sector')['raised_amount_usd'].transform('sum')
D2['Totcount'] = D2.groupby('main_sector')['raised_amount_usd'].transform('count')
D2

company_permalink funding_round_type  \
291                    /organization/365scores            venture   
629                     /organization/7digital            venture   
630                     /organization/7digital            venture   
684                        /organization/90min            venture   
853                      /organization/abcodia            venture   
1000               /organization/acacia-pharma            venture   
1020                 /organization/acal-energy            venture   
1023   /organization/acal-enterprise-solutions            venture   
1092            /organization/accent-media-ltd            venture   
1216           /organization/accumuli-security            venture   
1217                 /organization/accunostics            venture   
1375                /organization/acs-clothing            venture   
1376                /organization/acs-clothing            venture   
1743                     /organization/adbrain            venture   
1906                   /organization/adinsight            venture   
1927                       /organization/adjug            venture   
2335              /organization/aea-technology            venture   
2381                      /organization/aepona            venture   
2382                      /organization/aepona            venture   
2876                       /organization/aihit            venture   
2914           /organization/air-semiconductor            venture   
3379                    /organization/alfresco            venture   
3380                    /organization/alfresco            venture   
3704                     /organization/almondy            venture   
3750             /organization/alphamosaic-ltd            venture   
4077                        /organization/amee            venture   
4219        /organization/amino-communications            venture   
4295                   /organization/amplience            venture   
4296                   /organization/amplience            venture   
4297                   /organization/amplience            venture   
...                                        ...                ...   
85490                 /organization/weswap-com            venture   
85914                   /organization/winbuyer            venture   
86267                      /organization/wonga            venture   
86642                 /organization/wunderloop            venture   
86843                   /organization/xconnect            venture   
86844                   /organization/xconnect            venture   
86891        /organization/xenetic-biosciences            venture   
86912                    /organization/xention            venture   
86913                    /organization/xention            venture   
86914               /organization/xercise4less            venture   
86915               /organization/xercise4less            venture   
87059                       /organization/xmos            venture   
87067                        /organization/xo1            venture   
87363                /organization/yasa-motors            venture   
87419                   /organization/yd-world            venture   
87554                   /organization/yieldify            venture   
87838                       /organization/yoyo            venture   
87844                      /organization/yplan            venture   
88064                   /organization/zattikka            venture   
88090               /organization/zbd-displays            venture   
88409                      /organization/zesty            venture   
88447                       /organization/zeus            venture   
88448                       /organization/zeus            venture   
88499              /organization/ziarco-pharma            venture   
88607                    /organization/zinwave            venture   
88608                    /organization/zinwave            venture   
88848                     /

In [353]:
# Adding the sum and count as columns to D3
D3['Totval'] = D3.groupby('main_sector')['raised_amount_usd'].transform('sum')
D3['Totcount'] = D3.groupby('main_sector')['raised_amount_usd'].transform('count')
D3

company_permalink funding_round_type  \
0                                    /organization/-fame            venture   
164                       /organization/21diamonds-india            venture   
760                         /organization/a-little-world            venture   
1935                        /organization/adlabs-imagica            venture   
2683                                 /organization/agile            venture   
3973                      /organization/amagi-media-labs            venture   
4840                                /organization/apalya            venture   
4842                                /organization/apalya            venture   
5208                               /organization/applabs            venture   
5964                               /organization/ariisto            venture   
6622                          /organization/ather-energy            venture   
6787                      /organization/attero-recycling            venture   
6960                        /organization/aujas-networks            venture   
7432                            /organization/avesthagen            venture   
7855                     /organization/azure-hospitality            venture   
7857                           /organization/azure-power            venture   
7859                           /organization/azure-power            venture   
7860                           /organization/azure-power            venture   
7909                               /organization/babajob            venture   
8068                         /organization/bakers-circle            venture   
8175                            /organization/bankbazaar            venture   
8176                            /organization/bankbazaar            venture   
8985                         /organization/bestylish-com            venture   
9157   /organization/bharat-light-and-power-group-pri...            venture   
9158   /organization/bharat-light-and-power-group-pri...            venture   
9159                      /organization/bharat-matrimony            venture   
10798                        /organization/bluestone-com            venture   
11114                            /organization/bookmycab            venture   
13163         /organization/canvera-digital-technologies            venture   
13218                        /organization/capital-float            venture   
...                                                  ...                ...   
76224                      /organization/taxiforsure-com            venture   
76258                               /organization/teabox            venture   
76557                              /organization/teledna            venture   
77423                           /organization/the-porter            venture   
78336                   /organization/tinyowl-technology            venture   
78571                        /organization/tonbo-imaging            venture   
78676                                /organization/toppr            venture   
79434                        /organization/treebo-hotels            venture   
79754                          /organization/tripfactory            venture   
79873                 /organization/trivitron-healthcare            venture   
79944                          /organization/trucksfirst            venture   
80414                       /organization/tutorvista-com            venture   
81074                          /organization/unicommerce            venture   
81520                            /organization/urbanclap            venture   
81891                  /organization/valyoo-technologies            venture   
82216                          /organization/veeba-foods            venture   
83989                               /organization/vizury            venture   
84409                                /organization/vserv            venture   
84504                    /organization/vyome-biosciences            venture   
85152     

### Extracting solutions from D1, D2, D3

In [354]:
print('Total number of investments :' + str(D1[['main_sector','Totval','Totcount']].drop_duplicates()['Totcount'].sum()))
print('Total amount of investments :' + str(D1[['main_sector','Totval','Totcount']].drop_duplicates()['Totval'].sum()))
print(D1[['main_sector','Totval','Totcount']].drop_duplicates().sort_values('Totcount',ascending=False))



Total number of investments :9942
Total amount of investments :88638294664.0
                                 main_sector        Totval  Totcount
72                                    Others  2.173601e+10      2423
9    Social, Finance, Analytics, Advertising  1.939738e+10      2216
173               Cleantech / Semiconductors  1.780163e+10      1951
149               News, Search and Messaging  1.121957e+10      1280
654                                   Health  6.866859e+09       760
80                             Manufacturing  6.228553e+09       691
609                            Entertainment  4.119198e+09       479
331                      Automotive & Sports  1.269104e+09       142


In [355]:
# Companies that invested more in top and second top main_sector
print(D1.loc[D1[D1['main_sector']=='Others']['raised_amount_usd'].idxmax()]['name'])
print(D1.loc[D1[D1['main_sector']== 'Social, Finance, Analytics, Advertising']['raised_amount_usd'].idxmax()]['name'])

SpiderCloud Wireless
Flite


In [356]:
print('Total number of investments :' + str(D2[['main_sector','Totval','Totcount']].drop_duplicates()['Totcount'].sum()))
print('Total amount of investments :' + str(D2[['main_sector','Totval','Totcount']].drop_duplicates()['Totval'].sum()))
print(D2[['main_sector','Totval','Totcount']].drop_duplicates().sort_values('Totcount',ascending=False))



Total number of investments :559
Total amount of investments :4875228300.0
                                  main_sector        Totval  Totcount
1375                                   Others  1.123624e+09       129
853                Cleantech / Semiconductors  1.110140e+09       124
291   Social, Finance, Analytics, Advertising  9.744040e+08       118
1092               News, Search and Messaging  5.657462e+08        65
629                             Entertainment  4.227847e+08        48
4709                            Manufacturing  3.319403e+08        38
1217                                   Health  2.095375e+08        23
7190                      Automotive & Sports  1.370516e+08        14


In [357]:
# Companies that invested more in top and second top main_sector
print(D2.loc[D2[D2['main_sector']=='Others']['raised_amount_usd'].idxmax()]['name'])
print(D2.loc[D2[D2['main_sector']== 'Cleantech / Semiconductors']['raised_amount_usd'].idxmax()]['name'])

MyOptique Group
Onyvax


In [358]:
print('Total number of investments :' + str(D3[['main_sector','Totval','Totcount']].drop_duplicates()['Totcount'].sum()))
print('Total amount of investments :' + str(D3[['main_sector','Totval','Totcount']].drop_duplicates()['Totval'].sum()))
print(D3[['main_sector','Totval','Totcount']].drop_duplicates().sort_values('Totcount',ascending=False))

    

Total number of investments :243
Total amount of investments :2174543602.0
                                   main_sector       Totval  Totcount
164                                     Others  703409507.0        78
760    Social, Finance, Analytics, Advertising  385549550.0        45
7909                News, Search and Messaging  368834545.0        41
0                                Entertainment  200830000.0        23
8068                             Manufacturing  165900000.0        18
11114                                   Health  157740000.0        17
6787                Cleantech / Semiconductors  125380000.0        14
6622                       Automotive & Sports   66900000.0         7


In [359]:
# Companies that invested more in top and second top main_sector
print(D3.loc[D3[D3['main_sector']=='Others']['raised_amount_usd'].idxmax()]['name'])
print(D3.loc[D3[D3['main_sector']== 'Social, Finance, Analytics, Advertising']['raised_amount_usd'].idxmax()]['name'])


EximSoft-Trianz
inTarvo
